### ISSUE TITLE
Broken default value for Code/JSON fields

### ISSUE THREAD

**Issue by jacob418** [laravel/nova-issues#3736]
- Laravel Version: 8.75.0
- Nova Version: 3.30.0
- PHP Version: 8.0.9
- Database Driver & Version: Docker-Image: `postgres:13`
- Operating System and Version: Linux 5.4.0-96-generic
- Browser type and version: Google Chrome/96.0.4664.110

### Description:

For fields of type `\Laravel\Nova\Fields\Code` with the `mode` = `application/json` option it is not possible to define default values via the `\Laravel\Nova\Fields\Code::default` method.

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Create a Laravel/Nova Project
2. Create a Model/Table with JSON column
3. Create a Nova Resource for the Model
4. Add a `Code` field and set the JSON-Option
5. Set  `'{}'` as default value for the Field
6. Browser: open the 'create a new resource' form for the Model in Nova UI

**Expected behavior**
- Field gets default-value (`{}`) displayed as content

**Observed behavior**
- Field gets `null` displayed as content

### Reason

The problem originates in `\Laravel\Nova\Fields\Code::resolveAttribute`.
 When the JSON option is set, the current value is encoded as JSON in line 66/56.
 However, encoding `null` with JSON will result in the string `'null'`.
 
Later in the `\Laravel\Nova\Fields\Field::jsonSerialize` method, the default value is resolved *only* if the current value is nullish.
 However the string `'null'` is not nullish (even though the encoded value is nullish) so the default value will not be resolved.
 
### Workaround

A way to work around this by extending the code field with your own class:

*app/Nova/Fields/code.php*
```php
<?php

namespace App\Nova\Fields;

use Laravel\Nova\Fields\Field;

class Code extends \Laravel\Nova\Fields\Code
{
    protected function resolveAttribute($resource, $attribute): mixed
    {
        $value = Field::resolveAttribute($resource, $attribute);

        if ($this->json && !is_null($value)) {
            return json_encode($value, $this->jsonOptions ?? JSON_PRETTY_PRINT);
        }

        return $value;
    }
}
```

In line 20 we ensure the value to encode is not `null` before serializing it.
 However this introduces a problem in scenarios where a `null` value is expected to be inserted into db in certain cases so use the workaround with caution! :warning::warning::warning:

**Comment by crynobone** [comment]
Based on my understanding, the observed behavior is expected at the moment and this is a change request.

**Comment by davidhemphill** [comment]
👋 Hi there! In order to keep this repository focused on bug reports, we auto-close feature requests and requests for help. Feel free to post your feature requests so others can discuss and add reactions. We'll keep an eye on them for later planning.

**Comment by jacob418** [comment]
> Based on my understanding, the observed behavior is expected at the moment and this is a change request.

No, the expected behavior for a call to the `default` method is not for it to have NO effect without any warnings or notices. This is completely unexpected and leads to misunderstandings.

If this is intended it should be in the docs for the code field with a **BIG** exclamation mark! Also an alternative is to thrown an exception on use. Just ignoring the problem is no option in my opinion, if it is not taken care of it should be made clear to the developer in some way!
